src/v/iceberg/rest_client/catalog_client.cc

```diff
@@ -172,6 +184,61 @@ ss::future<expected<iobuf>> catalog_client::perform_request(
     }
 }
 
+ss::future<expected<load_table_result>> catalog_client::create_table(
+  const chunked_vector<ss::sstring>& ns,
+  create_table_request req,
+  retry_chain_node& rtc) {
+    auto token = co_await ensure_token(rtc);
+    if (!token.has_value()) {
+        co_return tl::unexpected(token.error());
+    }
+    auto http_request = table{root_path(), ns}
+                          .create()
+                          .with_bearer_auth(token.value())
+                          .with_content_type(json_content_type);
+
+    co_return (co_await perform_request(
+                 rtc, http_request, serialize_payload_as_json(req)))
+      .and_then(parse_json)
+      .and_then(parse_as_expected("create_table", parse_load_table_result));
+}
+
+ss::future<expected<load_table_result>> catalog_client::load_table(
+  const chunked_vector<ss::sstring>& ns,
+  const ss::sstring& table_name,
+  retry_chain_node& rtc) {
+    auto token = co_await ensure_token(rtc);
+    if (!token.has_value()) {
+        co_return tl::unexpected(token.error());
+    }
+
+    auto http_request
+      = table(root_path(), ns).get(table_name).with_bearer_auth(token.value());
+
+    co_return (co_await perform_request(rtc, http_request))
```

Comment:
I was looking at the related code out of curiosity and saw this

https://github.com/redpanda-data/redpanda/blob/dev/src/v/iceberg/rest_client/catalog_client.cc#L158

It looks like in the case of retries it copies the payload but the header is a std::move(request.value()) and value() returns an lvalue, so I think move of an lvalue casts it to a const value& but its a bit mind bending to guarantee that there is no use after move and I'm still not convinced, just that I'd post it while we are here.

(we should have some coverage for rest catalog client retries if there aren't already).

Reply:
good point, it should not move that value, i will add some retry tests